### PR TITLE
New Package: Ollama

### PR DESCRIPTION
**Pull request by Cheetah26**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.6 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.6.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/141891)

### CODE REVIEW

**Comment by og-mrk** [comment]
### Brief Introduction

Hello @Cheetah26,
I've noticed that this is your first PR to this Repo, and if so, welcome!
I new myself to Winget as well, but I'll share some tips/advices based on your PR, hope it'll guide you in the right direction.

**Note**: If you have any question, you can ask right ahead in this PR, or in [Winget's Repo Discussions](https://github.com/microsoft/winget-pkgs/discussions), either one is fine.

### Filling in the form

> * [ ]  Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
> * [ ]  Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
> * [ ]  This PR only modifies one (1) manifest
> * [ ]  Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
> * [ ]  Have you tested your manifest locally with `winget install --manifest <path>`?
> * [ ]  Does your manifest conform to the [1.6 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.6.0)?
> 
> Note: `<path>` is the name of the directory containing the manifest you're submitting.

It's good to fill in the Required Items in the PR's Template List (Put an `x` between `[ ]` to check an item).

the first one is you, as a person, organization, Company, and/or entity agreeing on `Contributor License Agreement`, or "cla" for short, you can read what it states in [this link](https://cla.opensource.microsoft.com/microsoft/winget-pkgs).

The second one is very important, which's checking if your PR changes One Manifest, or not, because double checking this in your PR's Changes is going to save you, and maintainers time, mainly because Maintainers needs to see all the Automated Checks Pass, and _currently_ it only support One Manifest for every PR. Therefore, changes in Multiple Manifests are Not Allowed, the tests will fail, and your PR will not be merged.

The Third and Forth are, in general, optional ones, but if you have the time to test the manifest that you've Updated/Created, then it'll add some confidence in your PR 😄. [Read more over here](https://github.com/microsoft/winget-pkgs/blob/master/README.md#test-your-manifest)

Last one is an easy check, because when you use tools like `wingetcreate` or `YAMLCreate.ps1` in the `Tools` Folder, these tools will use the correct manifest version. So make sure to check it as well.

### Test if your install is Completely Silent (Unattended Install)

From what I see in your manifest, the package will simply run `OllamaSetup.exe` with **No Installer Switches** passed to it, which's already a Big Red Flag.
you should first see if the installation process for an app you want to add to winget Can Be converted from "Interactive Mode" (The user can install said app Interactively), to an "Unattended Mode" (One Command to run, and everything is installed, **No User Input Required**).

### How to know if I can do the what I've explained?

First, download the installer onto your computer (for this case, it's `OllamaSetup.exe`), then open up a Terminal (CMD or PowerShell) inside the Folder which contains the installer, then run `APP-NAME.exe /?` or `APP-NAME.exe --help`, for your case, it should be `OllamaSetup.exe /?` or `OllamaSetup.exe --help` (Either one should be fine), and a Window will popup, telling you All of the available switches you can pass to the installer.

### Screenshot for reference

image

**Comment by og-mrk** [comment]
> Url validation error
> 
> * manifests\o\Ollama\Ollama\0.1.27
>   
>   * https://ollama.com/download/OllamaSetup.exe
>     
>     * This URL has previously been reported as Malware to [Microsoft Defender SmartScreen](https://docs.microsoft.com/windows/security/threat-protection/microsoft-defender-smartscreen/microsoft-defender-smartscreen-overview)
>     * If you believe this is in error, you may [file a report](https://feedback.smartscreen.microsoft.com/feedback.aspx?v=6&t=16&result=block&type=top&url=aHR0cHM6Ly9vbGxhbWEuY29tL2Rvd25sb2FkL09sbGFtYVNldHVwLmV4ZQ%3D%3D)

Also @Cheetah26, using URLs outside of Github or SourceForge (For Example) is ok, but if the Validation Tests might have some False Positives (a URL that this test thinks is a Malicious One... but in reality there's nothing wrong with it).

I've tested this particular URL myself, and it download a `OllamaSetup.exe` file, and I can save it on my PC, and looking at the Digital Signature and Hash Values of this downloaded file, it matches _exactly_ to the one provided by [Ollama's Github Releases Page](https://github.com/ollama/ollama/releases/).

Of course this False Positive is somewhat rare to occur, but it's recommended to use the Github Link One for two reasons:
* It's hosted on a Trusted Website (github.com)
* When you provide the Install Link for this version, it'll be a Static One, meaning in 10 years from now, the same link will return the same file, no matter what. (https://github.com/ollama/ollama/releases/download/v0.1.27/OllamaSetup.exe  <-- Notice the `v0.1.27` in the link)
While the one you've added will return different versions of OllamaSetup.exe, overtime...

**Comment by Cheetah26** [comment]
Thanks for all the tips!

I think I've got it fixed up.  Let me know if any more changes are needed.

File: manifests/o/Ollama/Ollama/0.1.27/Ollama.Ollama.installer.yaml
Comment (og-mrk) [inline manifests/o/Ollama/Ollama/0.1.27/Ollama.Ollama.installer.yaml:1]: Good job on the changes 👍

**Comment by og-mrk** [comment]
> Thanks for all the tips!
> 
> I think I've got it fixed up. Let me know if any more changes are needed.

no problem @Cheetah26, happy to help.

One last tip of mine, which really helps if you're planning on Making New Manifests (Add New Packages, or New Versions of a Package), or Updating the Metadata of a Manifest.

### One More Byte?
The tip I'll be explaining in Q&A style, hope it isn't verbose/annoying, and if you've any question, you can ask right away!

What're all of the Available Fields I can add to a Manifest? (Ones that `wingetcreate` doesn't prompt you for, because they're optional Metadata)

To answer this, you need to know What Manifest Version are you using? <-- See the `ManifestVersion` Field that's found in Each File for a Manifest. Which for the last few months, it's version `1.6.0`
Then go into the [docs folder](https://github.com/microsoft/winget-pkgs/tree/master/doc) in winget-pkgs repo, over there you'll find two folders, one for [Tools](https://github.com/microsoft/winget-pkgs/tree/master/doc/tools) and another [Manifest/Schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema) Documentation, inside the `Manifest/Schema` Folder, you'll find Documentation for Every Version! even future one being prepared is well documented! (version 1.7.0)
Then you can see what are the available field.

> [!NOTE]
> Adding Extra Metadata (More Fields) is recommended, but not required.
> It'll make the Packages more "complete" in the sense of providing more information to display to the User.

**Review by og-mrk — COMMENTED** [review]
Hi @Cheetah26,
Good job on the changes 👍
looks like the validation tests has failed again, but don't worries, we can try again and see if we can fix this issue 😄

Now looking back at the available options, two have popped-up to my mind, which's `/SP-` and `/SUPPRESSMSGBOXES`, **_see the screenshot below_**.

image

Try adding these two to the sub-field `custom` under `InstallerSwitches`, **_see the screenshot below_**.

image

File: manifests/o/Ollama/Ollama/0.1.27/Ollama.Ollama.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+PackageIdentifier: Ollama.Ollama
+PackageVersion: 0.1.27
+InstallerType: inno
+InstallModes:
+  - silent
+  - silentWithProgress
+InstallerSwitches:
+  Silent: /VERYSILENT
+  SilentWithProgress: /SILENT
```
Comment (og-mrk) [inline manifests/o/Ollama/Ollama/0.1.27/Ollama.Ollama.installer.yaml:12]:
```suggestion
InstallerSwitches:
  Silent: /VERYSILENT
  SilentWithProgress: /SILENT
  Custom: /SP- /SUPPRESSMSGBOXES
```

**Comment by stephengillie** [comment]
Automatic Validation ended with:
> No errors to post.

(Automated response - build 857.)

File: manifests/o/Ollama/Ollama/0.1.27/Ollama.Ollama.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+Installers:
+  - Architecture: x86
```
Comment (stephengillie) [inline manifests/o/Ollama/Ollama/0.1.27/Ollama.Ollama.installer.yaml:14]:
```suggestion
  - Architecture: x64
```

**Review by stephengillie — CHANGES_REQUESTED** [review]
Hi @Cheetah26,

This package appears to be 64-bit. Our `x86` validations are performed on 32-bit OSes.

**Review by stephengillie — APPROVED** [review]

**Pull request merged**